src/Compilers/VisualBasic/Portable/Emit/PEModuleBuilder.vb

```diff
@@ -641,5 +645,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Emit
             Return _disableJITOptimization.ContainsKey(methodSymbol)
         End Function
 
+        Friend Function GetModuleVersionId(mvidType As Cci.ITypeReference, syntaxOpt As VisualBasicSyntaxNode, diagnostics As DiagnosticBag) As Cci.IFieldReference
+            Dim details As PrivateImplementationDetails = GetPrivateImplClass(syntaxOpt, diagnostics)
+            EnsurePrivateImplementationDetailsStaticConstructor(details, syntaxOpt, diagnostics)
+
+            Return details.GetModuleVersionId(mvidType)
+        End Function
+
+        Friend Function GetInstrumentationPayloadRoot(analysisKind As Integer, payloadType As Cci.ITypeReference, syntaxOpt As VisualBasicSyntaxNode, diagnostics As DiagnosticBag) As Cci.IFieldReference
+            Dim details As PrivateImplementationDetails = GetPrivateImplClass(syntaxOpt, diagnostics)
+            EnsurePrivateImplementationDetailsStaticConstructor(details, syntaxOpt, diagnostics)
+
+            Return details.GetOrAddInstrumentationPayloadRoot(analysisKind, payloadType)
+        End Function
+
+        Private Sub EnsurePrivateImplementationDetailsStaticConstructor(details As PrivateImplementationDetails, syntaxOpt As VisualBasicSyntaxNode, diagnostics As DiagnosticBag)
+            If details.GetMethod(WellKnownMemberNames.StaticConstructorName) Is Nothing Then
+                details.TryAddSynthesizedMethod(New SynthesizedPrivateImplementationDetailsSharedConstructor(SourceModule, details, GetUntranslatedSpecialType(SpecialType.System_Void, syntaxOpt, diagnostics)))
+            End If
+        End Sub
```

Comment:
`PrivateImplementaionDetails` is a class shared between C# and VB. I was imagining all three methods moving to `CommonPEModuleBuilder` almost unchanged (`VisualBasicSyntaxNode` would become `SyntaxNode` perhaps). The one interesting change would be adding the `CreateStaticConstructor` abstract method that `EnsurePrivateImplementationDetailsStaticConstructor` would call. But it was just a suggestion and not important enough to block this PR.